### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by soundslocke**
Improve HTTP headers by:

- Converting headers to lowercase for easier handling
- Avoid extraneous PHP version checks
- Remove extraneous header function from Client class

This reworks [this previous fix](https://github.com/Bandwidth/php-bandwidth/commit/5e1cd10dbab408193604a2ed19e20a7aafd3f810) in a more foolproof way.

Also improve composer.json to have consistent spacing.

### CODE REVIEW

File: source/utils/Locator.php [hunk 1]
```diff
@@ -26,23 +26,14 @@ class Locator extends BaseUtilities {
...
-      $header = null;
-      $phpver = explode('.', phpversion());
-
-      if ($phpver[0] >= 7 && $phpver[1] >= 2) {
-        $header = parent::find($headers, "location");
-      }
-      else {
-        $header = parent::find($headers, "Location");
-      }
+      $header = parent::find($headers, "Location");
```
Comment (jmulford-bandwidth) [inline source/utils/Locator.php]: Thanks for the contribution! This is still causing issues with PHP versions < 7.2

```
jmulford@jmulford-mbp php-projects $ /usr/bin/php test-script.php 

Fatal error: Uncaught Exception: No header found as location:
  thrown in /Users/jmulford/projects/php-projects/vendor/bandwidth/catapult/source/utils/Base.php on line 59
jmulford@jmulford-mbp php-projects $ /usr/bin/php -v
PHP 7.1.23 (cli) (built: Nov 27 2018 16:59:25) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
```

It looks like `Location` is the value being set in the header for php < 7.2. I'm investigating this but if you have any changes feel free to add to this PR

**Comment by jmulford-bandwidth** [comment]
Also the travis script is not properly setup so don't worry about that if it fails. I'm investigating changes to that as well

File: source/utils/Locator.php (hunk 1 above)
Comment (soundslocke) [reply source/utils/Locator.php]: The [place where I convert all the headers to lowercase](https://github.com/Bandwidth/php-bandwidth/pull/39/commits/091092cb10b68b59c1bb87b28e409b5ab399e09b#diff-612961465cc4e3ace26669325dd222e9R344) must not be the only place headers get set? Or I'm missing something else. Thanks for the heads up, I certainly didn't mean to break older PHP support.

File: source/core/Client.php [hunk 2]
```diff
@@ -350,9 +339,13 @@ private function request($method, $url, $data=array(), $mixed=FALSE, $decode=FAL
           $headers[$h[0]] = trim(str_replace("\r\n", "", str_replace("\n", "", $h[1]))); 
       }
 
+      /* HTTP headers are case-insensitive so let's convert them to lowercase */
+      /* for internal consistency when working with them */
+      $headers = array_map('strtolower', $headers);
```
Comment (jmulford-bandwidth) [inline source/core/Client.php]: Try changing this line to

```
      $headers = array_change_key_case($headers, CASE_LOWER);
```

That worked for me

File: source/utils/Locator.php (hunk 1 above)
Comment (jmulford-bandwidth) [reply source/utils/Locator.php]: I think you were supposed to use this method

https://php.net/manual/en/function.array-change-key-case.php

That change worked for me. If that works for you feel free to make that change and we will merge this after we review

Comment (soundslocke) [reply source/utils/Locator.php]: Thanks for getting to it before me! That is indeed my mistake: the headers are the array keys, not the values :sweat_smile: 

Will push a change to this fork soon.

Comment (soundslocke) [reply source/utils/Locator.php]: I have adjusted the fork accordingly. Feel free to re-review whenever.

Comment (jmulford-bandwidth) [reply source/utils/Locator.php]: Sweet. Looks good to me. Thank you!

**Pull request merged**